decoder.go

```diff
@@ -21,6 +21,17 @@ var (
 	nodeType reflect.Type = findNodeType()
 )
 
+// Unmarshal accepts a byte slice as input and writes the
+// data to the value pointed to by v.
+func Unmarshal(bs []byte, v interface{}) error {
```

Comment:
Is there an argument here for reordering the parameters as in Decode below? Practically it is Decode which doesn't honour the convention but that is in pretty heavy use now.